staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -43,65 +45,95 @@ type FeatureSupportChecker interface {
 	// initialzation, it can flip the support at runtime.
-	CheckClient(ctx context.Context, c client, feature storage.Feature) error
+	CheckClient(ctx context.Context, c client, feature storage.Feature)
 }
 
 type defaultFeatureSupportChecker struct {
 	lock                       sync.Mutex
 	progressNotifySupported    *bool
 	progresNotifyEndpointCache map[string]bool
+	checkingEndpoint           map[string]struct{}
 }
 
 func newDefaultFeatureSupportChecker() *defaultFeatureSupportChecker {
 	return &defaultFeatureSupportChecker{
 		progresNotifyEndpointCache: make(map[string]bool),
+		checkingEndpoint:           make(map[string]struct{}),
 	}
 }
 
 // Supports can check the featue from anywhere without storage if it was cached before.
-func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool, error) {
+func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) bool {
 	switch feature {
 	case storage.RequestWatchProgress:
 		f.lock.Lock()
 		defer f.lock.Unlock()
 
-		return ptr.Deref(f.progressNotifySupported, false), nil
+		return ptr.Deref(f.progressNotifySupported, false)
 	default:
-		return false, fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
+		klog.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
+		return false
 	}
 }
 
 // CheckClient accepts client and calculate the support per endpoint and caches it.
-// It will return at any point if error happens or one endpoint is not supported.
-func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
+func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		f.checkClient(ctx, c)
 	default:
-		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
-
+		klog.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, c client) {
+	// start with 10 ms, multiply by 2 each step, 16 steps ~= 10 minutes
+	backoff := wait.Backoff{
+		Duration: 10 * time.Millisecond,
+		Jitter:   0,
+		Factor:   2,
+		Steps:    16,
+	}
 	f.lock.Lock()
 	defer f.lock.Unlock()
-
 	for _, ep := range c.Endpoints() {
-		supported, err := f.supportsProgressNotifyEndpointLocked(ctx, c, ep)
-		if err != nil {
-			return err
-		}
-		if !supported {
-			f.progressNotifySupported = ptr.To(false)
-			return nil
+		if _, found := f.checkingEndpoint[ep]; found {
+			continue
 		}
+		f.checkingEndpoint[ep] = struct{}{}
+		go func(ep string) {
+			err := wait.ExponentialBackoffWithContext(ctx, backoff, func(ctx context.Context) (done bool, err error) {
```

Comment:
Add `defer utilruntime.HandleCrash()`

Reply:
Just changed.

Reply:
`defer utilruntime.HandleCrash()` catches panic from a goroutine and prevents the whole server from crashing. I think you should add it.

Reply:
I didn't mean to replace `wait.ExponentialBackoffWithContext` :)

Reply:
Thanks i'll add it
> I didn't mean to replace wait.ExponentialBackoffWithContext :)

I just pushed the code and saw your'e message, i meant to check again. 👍